### DISCUSSION TITLE
Where type hierarchy resolution and serialization/deserialization happens? I need to implement F# DU support

### DISCUSSION THREAD

**Discussion by xperiandri** [Gremlinq/ExRam.Gremlinq#1078] [Q&A] [answered]
F# discriminated union case is either a subclass or a static property value (if no parameters).
I need to save that as a class hierarchy.
Where in the Gremlinq code to look at?

Related #66

**Comment by danielcweber** [comment]
I have a slight clue of what you're asking for....a pull request with a unit test that contains failing assertions that roughly show what the expected outcome of the proposed feature would be very helpful though.

**Reply by xperiandri** [reply, marked as answer]
Do you mean this?
https://github.com/ExRam/ExRam.Gremlinq/blob/2ffcaba2b52b27df2b5bcdc6da5cdb3d2ec2c432/test/ExRam.Gremlinq.Support.NewtonsoftJson.Tests/RegisterNativeTypeTests.cs

**Reply by xperiandri** [reply]
Could you point to an object registration test? Where I can see object serialization in a custom way
